### ISSUE TITLE
IR Outliner: Assertion `V != nullptr && "Value is a nullptr?"

### ISSUE THREAD

**Issue by ornata** [llvm/llvm-project#55078]
https://godbolt.org/z/Y31a7cqPd

```
opt: /root/llvm-project/llvm/include/llvm/Analysis/IRSimilarityIdentifier.h:896: llvm::Optional<unsigned int> llvm::IRSimilarity::IRSimilarityCandidate::getGVN(llvm::Value*): Assertion `V != nullptr && "Value is a nullptr?"' failed.
```

Bugpointed from building `llvm-test-suite/tools/fpcmp.c` with the IR outliner at -O2 for AArch64.

**Comment by ornata** [comment]
@AndrewLitteken

**Comment by ornata** [comment]
(This also seems to happen when building `MultiSource/Applications/SPASS` at -Oz with the IR outliner enabled.)

**Comment by serge-sans-paille** [comment]
This got fixed by e38f014c40e94205925403fa75e48244ee31b6c0
